### PR TITLE
Fix failed to persist/unpersist database and add tests

### PR DESCRIPTION
**Pull request by qnkhuat**
The bug: triggering persist/unpersist a database will result in a 500 call with error message:
` Assert failed: (sequential? sql-args)`

The cause of this is from here https://github.com/metabase/metabase/blob/821d2f0310f731942f7cd5d10815437236eb0b5e/src/metabase/api/database.clj#L843, the `t2/update!` method accepts changes as a map, not key-value pair. So it should be `(t2/update! Database id {:options ...})`

Introduced by https://github.com/metabase/metabase/pull/29308. I did another round of checking and I think this is the only place left where we made that mistake.

This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29833)

### CODE REVIEW

**Review by dpsutton — APPROVED** [review]

**Review by dpsutton — DISMISSED** [review]
seems like all of the tests are quite unhappy.

**Review by calherries — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29833)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @dpsutton and @qnkhuat)

___
*[`test/metabase/api/database_test.clj` line 1517 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPDKQtABKeE5_BANRR:-NSPDKQtABKeE5_BANRS:b6s9ycw) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1517)):*
> ```Clojure
>         (t2.with-temp/with-temp
>           [Card card {:database_id db-id
>                       :dataset     true}]
> ```

nit: I think the convention is to put the bindings to the right, like a let block.



_Suggestion:_
```Clojure
        (t2.with-temp/with-temp [Card card {:database_id db-id
        ... 
```
___
*[`test/metabase/api/database_test.clj` line 1533 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPCl2P0VIm2BA4FiKH:-NSPCl2P0VIm2BA4FiKI:bg1ul86) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1533)):*
> ```Clojure
>                                                   :database_id db-id
>                                                   :card_id     (:id card))))
>               (is (= true (t2/select-one-fn (comp :persist-models-enabled :options)
> ```

These make for nicer error messages



_Suggestion:_
```Clojure
              (is (true? (t2/select-one-fn (comp :persist-models-enabled :options)
```
___
*[`test/metabase/api/database_test.clj` line 1554 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPCU-V28BFffRFbfcN:-NSPCU-V28BFffRFbfcO:undefined) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1554)):*
> ```Clojure
>               ;; trigger persist first
>               (mt/user-http-request :crowberto :post 204 (str "database/" db-id "/unpersist"))
>               (is (= nil (t2/select-one-fn (comp :persist-models-enabled :options)
> ```





_Suggestion:_
```Clojure
              (is (nil? (t2/select-one-fn (comp :persist-models-enabled :options)
```

**Review by calherries — COMMENTED** [review]
Looks good, just left some tiny aesthetic suggestions

> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29833#-:-NSPDoIG1z78QFtN_K95:bhn18zk)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @dpsutton and @qnkhuat)

**Review by qnkhuat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29833)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @calherries and @dpsutton)

___
*[`test/metabase/api/database_test.clj` line 1517 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPDKQtABKeE5_BANRR:-NSPKkl99gccKfOVhguv:behkiin) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1517)):*
> nit: I think the convention is to put the bindings to the right, like a let block.
is there any benefit to it? I feel like it makes the binding block much longer.
___
*[`test/metabase/api/database_test.clj` line 1533 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPCl2P0VIm2BA4FiKH:-NSPL5dZ5Aph-JiSuclh:b-896fix) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1533)):*
<details><summary><i>Previously, calherries (Cal Herries) wrote…</i></summary><blockquote>

These make for nicer error messages
</blockquote></details>

Done.
___
*[`test/metabase/api/database_test.clj` line 1554 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPCU-V28BFffRFbfcN:-NSPL5Bh6HcjQ36zC9Vm:b-896fix) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1554)):*
> ```Clojure
>               ;; trigger persist first
>               (mt/user-http-request :crowberto :post 204 (str "database/" db-id "/unpersist"))
>               (is (= nil (t2/select-one-fn (comp :persist-models-enabled :options)
> ```

Done.

**Comment by calherries** [comment]
*[`test/metabase/api/database_test.clj` line 1517 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPDKQtABKeE5_BANRR:-NSPVZIkDMAHL4kGZVeL:bg3ziv8) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1517)):*
<details><summary><i>Previously, qnkhuat (Ngoc Khuat) wrote…</i></summary><blockquote>

> nit: I think the convention is to put the bindings to the right, like a let block.
is there any benefit to it? I feel like it makes the binding block much longer.
</blockquote></details>

The only benefit from style consistency is that it can arguably aid reading comprehension. Imagine if 
I started 
writing all my messages with
new lines and whitespace


where 
you
didn't expect it. Its going to
slow your reading down 
and make every so slightly harder to find mistakes

**Review by qnkhuat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29833)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @calherries and @dpsutton)

___
*[`test/metabase/api/database_test.clj` line 1517 at r3](https://reviewable.io/reviews/metabase/metabase/29833#-NSPDKQtABKeE5_BANRR:-NSPWdvOC0Ecl9u3MFKY:b-pfnvy) ([raw file](https://github.com/metabase/metabase/blob/b4a755e8043dca877f3972b387e0f787846a30c2/test/metabase/api/database_test.clj#L1517)):*
<details><summary><i>Previously, calherries (Cal Herries) wrote…</i></summary><blockquote>

The only benefit from style consistency is that it can arguably aid reading comprehension. Imagine if 
I started 
writing all my messages with
new lines and whitespace


where 
you
didn't expect it. Its going to
slow your reading down 
and make every so slightly harder to find mistakes
</blockquote></details>

ha, that does make sense, but I'm not sure if it's absolutely necessary. for example, this test https://github.com/metabase/metabase/blob/a492e28301cc8431c8cbf2988a50a7cd8b725bf0/test/metabase/api/dashboard_test.clj#L231 if we use that convention then the line is gonna be ~180 chars.

not really readable IMO...

Another thing is the new `t2.with-temp/with-temp` is both longer and uses 3 arities instead of 2 now, so if you want to keep the vertical alignment of each arity, it gets even longer...

So Imma go with the current one.

**Review by calherries — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29833#-:-NSPa9HR0mfddFXV1pDx:bnfp4nl)* status: 0 of 3 files reviewed, all discussions resolved (waiting on @dpsutton)

File: .github/actions/test-driver/action.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ runs:
     - name: Build static viz frontend
       run: yarn build-static-viz
       shell: bash
+    - name: Setup upterm session
```
Comment (dpsutton) [inline .github/actions/test-driver/action.yml:21]: did you intend for this to be committed?

**Review by dpsutton — DISMISSED** [review]
> Setup upterm session

The changes look good but I think this addition in the action.yml needs to be justified or removed.

File: .github/actions/test-driver/action.yml (hunk 1 above)
Comment (qnkhuat) [reply .github/actions/test-driver/action.yml:21]: no, I was debugging the failing redshift tests. will remove

File: .github/actions/test-driver/action.yml [hunk 2]
```diff
@@ -18,6 +18,8 @@ runs:
...
+      uses: lhotari/action-upterm@v1
```
Comment (qnkhuat) [inline .github/actions/test-driver/action.yml:22]:
```suggestion
```

**Pull request merged**